test/foundry/fork/base/TestNexusSwapETHIntegration.t.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "./BaseSettings.t.sol";
+import "../../utils/Imports.sol";
+import "../../utils/NexusTest_Base.t.sol";
+import "../../shared/interfaces/IUniswapV2Router02.t.sol";
+import "../../shared/interfaces/IERC20.t.sol";
+
+/// @title TestNexusSwapETHIntegration
+/// @notice Tests Nexus smart account functionalities with Uniswap V2 swaps
+contract TestNexusSwapETHIntegration is NexusTest_Base, BaseSettings {
+    address payable private preComputedAddress;
+    IUniswapV2Router02 public uniswapV2Router;
+    MockPaymaster private paymaster;
```

Comment:
can use SamplePaymaster at least which is in infinitism samples/VerifyingPaymaster